extensions/binding/org.eclipse.smarthome.binding.hue/src/main/java/org/eclipse/smarthome/binding/hue/handler/HueLightHandler.java

```diff
@@ -486,4 +487,14 @@ private int getAlertDuration(Command command) {
 
         return delay;
     }
+
+    @Override
+    public void bridgeStatusChanged(ThingStatusInfo bridgeStatusInfo) {
+        super.bridgeStatusChanged(bridgeStatusInfo);
+        if (ThingStatus.ONLINE.equals(bridgeStatusInfo.getStatus())
+                && ThingStatus.OFFLINE.equals(getThing().getStatusInfo().getStatus())
```

Comment:
Do we really have to check all of this here or wouldn't it be enough to (always) act if the bridge switched to ONLINE?

Reply:
What if the thing state changed after it was set to offline because of the bridge going offline?

Reply:
Can this happen? We said that the only valid status for the Thing is OFFLINE.BRIDGE_OFFLINE while the bridge is offline. So this here should be the first moment when this changes.

Reply:
That is true. What happens if the bridge is OFFLINE for some reason and the thing goes OFFLINE.BRIDGE_OFFLINE, too. Then the user changes the configuration of the thing and sets an invalid configuration. The thing status might change to OFFLINE.CONFIGURATION_ERROR.

Reply:
Valid remark, this is all nasty :-/
But if we say that it is ok to have the Thing being OFFLINE for another reason than BRIDGE_OFFLINE when a bridge is OFFLINE, this also means that the BaseThingHandler implementation should only switch to OFFLINE.BRIDGE_OFFLINE, if it isn't OFFLINE for another reason already - BUT if this other reason is "COMMUNICATION_ERROR" changing it to BRIDGE_OFFLINE would imho be the correct thing, while OFFLINE.CONFIGURATION_ERROR should stay. I really don't know how to solve this best...

@maggu2810 & @sjka Any good ideas from you?

Reply:
* Bridge is offline
* Thing is offline
* Thing Status detail is ... (e.g. bridge offline or communication error or whatever)
* Bridge status changes to offline

Wouldn't it make sense (because it should succeed in the most cases) to handle the status transition similar as on initialization?
So, e.g. check configuration and if it does not fit, set offline:communication error, ...
I assume initialize and handling the bridge status change from offline to online could share some code.

I don't think we could rely on one status detail is more prioritized then the other one in offline state.

Reply:
Ok, so in @J-N-K's example above, it would mean that even if the status might have changed due to an invalid configuration, we would call the very same logic here, which simply should in any case re-check the validity of the configuration and do a best-effort approach to determine the current online status.
The result would be a more complex implementation than the current one (here in the HueLightHandler), but a more consistent one across bindings, correct?

Reply:
I think so.
Does it need more complex in the hue binding? Do we break anything if we simply call `void initializeThing(ThingStatus bridgeStatus)`?

Reply:
The more I think about it, probably BaseThingHandler should be abstract and each binding should find the correct way how to deal with bridgeStatusChanged.

Reply:
just curious is there a reason to go for equals instead of == for enums I would personally always prefer ==

Reply:
I don't think there is a reason but personal preference.
Enums are singletons so there shouldn't be any difference.

Reply:
> Does it need more complex in the hue binding?

Yes, because at the moment, it simply [switches to ONLINE](https://github.com/eclipse/smarthome/pull/4086/files/feb025795bec72ddec485c3c999ff1a60ac69d9e#diff-2ba4d1bdb2c89e2c8b66b342a88ed23eR497) and does not check whether it has a valid configuration, nor if it is (still) known/reachable on the hue bridge.

> Do we break anything if we simply call void initializeThing(ThingStatus bridgeStatus)?

For the Hue binding, this would probably be exactly the right thing to do. But this method only exists in that handler, so we cannot generalise it - we can only point to it as a good way to deal with it. But it means we have not formalized it and as such we are prone to implementation errors by binding developers.

> The more I think about it, probably BaseThingHandler should be abstract and each binding should find the correct way how to deal with bridgeStatusChanged.

Maybe that is the solution in the end.
@sjka Would appreciate your opinion here as well - maybe you have a good idea that nobody else thought of, who knows...

Reply:
> For the Hue binding, this would probably be exactly the right thing to do. But this method only exists in that handler, so we cannot generalise it - we can only point to it as a good way to deal with it. But it means we have not formalized it and as such we are prone to implementation errors by binding developers.

My comment has been only about the Hue binding.

Reply:
@kaikreuzer, `abstract` might not be a good idea, that way even binding developers that don't use brigdes at all would need to implement it. 

So, how do we go on from here?